src/storage/src/sink/kafka.rs

```diff
@@ -847,16 +807,15 @@ impl KafkaSinkState {
         &self,
         transaction_id: Timestamp,
         progress: &ProgressRunningState,
-    ) -> Result<(), anyhow::Error> {
+    ) {
         let encoded = serde_json::to_vec(&ProgressRecord {
             timestamp: transaction_id,
-        })?;
+        })
+        .expect("serialization to vec cannot fail");
         let record = BaseRecord::to(&progress.topic)
             .payload(&encoded)
             .key(&progress.key);
-        self.send(record).await?;
-
-        Ok(())
+        self.send(record).await;
```

Comment:
similar suggestion here! `let () = self.send(record).await;`? so we fail to compile if `send` changes to return a `Result` / some information on the `Ok` case.  I have the same suggestion for the other `send` calls throughout.